Sources/FluentKit/Utilities/GregorianDate.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+
+enum GregorianDateError: Error {
+    case invalidStringFormat
+}
+
+/**
+ A Gregorian date with no time or timezone, intended to pair with
+ the SQL `DATE` type
+ */
+public struct GregorianDate {
+    private static let gregorianCalendar = Calendar(identifier: .gregorian)
+    
+    public var year: Int
+    public var month: Int
+    public var day: Int
+    
+    public var dateComponents: DateComponents {
+        return DateComponents(calendar: GregorianDate.gregorianCalendar,
+                       year: year,
+                       month: month,
+                       day: day)
+    }
+    
+    /**
+     Create a GregorianDate from the current date in the current timezone.
+     */
+    public init() {
+        self.init(Date())
+    }
+    
+    /**
+     Create a GregorianDate from a year, month, and day.
+     The inputs are not validated to ensure they form a valid date.
+     
+     - Parameters:
+         - year: >= 0
+         - month: 1 – 12
+         - day: 1 – 31
+     */
+    public init(year: Int, month: Int, day: Int) {
+        self.year = year
+        self.month = month
+        self.day = day
+    }
+    
+    /**
+     Create a GregorianDate from the year, month, and day in the
+     DateComponents. Other fields in the components are ignored.
+     */
+    public init?(_ components: DateComponents) {
+        guard let year = components.year,
+            let month = components.month,
+            let day = components.day else { return nil }
+        
+        self.year = year
+        self.month = month
+        self.day = day
+    }
+    
+    /**
+     Create a GregorianDate from the date portion of a Date object,
+     using the timezone stored in that Date.
+     */
+    public init(_ date: Date) {
+        let components = GregorianDate.gregorianCalendar.dateComponents([.year, .month, .day], from: date)
+        self.year = components.year!
```

Comment:
Can we ever hit assertion for these force unwraps? If so, this init could be nil fallible.